src/tools/tidy/src/features.rs

```diff
@@ -172,7 +172,7 @@ pub fn check(path: &Path, bad: &mut bool) {
         "use_extern_macros", "staged_api", "const_indexing",
         "unboxed_closures", "stmt_expr_attributes",
         "cfg_target_thread_local", "unwind_attributes",
-        "inclusive_range_syntax"
+        "inclusive_range_syntax", "proc_macro"
```

Comment:
*Adding* entries to this list is not wanted, only removing them. I see though that you actually added a gate test, just only in the `compile-fail-fulldeps` directory, and the tidy check didn't run through that directory. So you only worked around a bug/missing feature in the tidy check, I'd say this is okay. I'll file a PR to remove it from the whitelist again.